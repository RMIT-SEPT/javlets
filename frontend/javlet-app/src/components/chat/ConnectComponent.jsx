import React, { Component } from "react";
import cookie from "js-cookie";
import axios from "axios";
import { API } from "../../Constants.js";

class ConnectComponent extends Component {
  constructor(props) {
    super(props);
    this.state = {};
    this.handleMessage = this.handleMessage.bind(this);
    this.handleRequest = this.handleRequest.bind(this);
  }

  render() {
    let button;

    switch (this.props.type) {
      case 0:
        button = (
          <button
            className="w3-btn w3-blue btn-primary btn-lg btn-block"
            type="button"
            onClick={this.handleMessage}
            value="Message"
          >
            Message
          </button>
        );
        break;
      case 1:
        button = (
          <button
            className="w3-btn w3-red btn-primary btn-lg btn-block"
            type="button"
            onClick={this.handleRequest}
            value="Add"
          >
            Request
          </button>
        );
        break;
      case 2:
        button = (
          <span className="w3-grey btn-primary btn-lg btn-block">Sent</span>
        );
        break;
      case 3:
        button = (
          <button
            className="w3-btn w3-green btn-primary btn-lg btn-block"
            type="button"
            onClick={this.handleRequest}
            value="Add"
          >
            Accept
          </button>
        );
        break;
      default:
        break;
    }

    return (
      <div className="connection">
        {button}
        <h4 className="w3-panel w3-blue">{this.props.name}</h4>
        <span className="w3-tiny">{this.props.conID}</span>
      </div>
    );
  }

  handleRequest(event) {
    const newItem = {
      senderId: cookie.get("id"),
      recipientId: this.props.conID
    };

    return axios.post(API + "/connection/add", newItem);
  }

  handleMessage(event) {
    cookie.set("rID", this.props.conID);
  }
}
export default ConnectComponent;
