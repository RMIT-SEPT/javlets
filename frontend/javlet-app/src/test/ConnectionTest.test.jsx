import React from 'react';
import { shallow } from 'enzyme';
import ConnectComponent from '../components/chat/ConnectComponent.jsx';

  it('ConnectComponent SNAPSHOTt', () => {

    // Test type 0
    let component = shallow(<ConnectComponent key={Math.random()} name="John" conID="s1111111" type={0}/>);

    // Test type 1
    component += shallow(<ConnectComponent key={Math.random()} name="Jill" conID="s1111111" type={1}/>);

    // Test type 2
    component += shallow(<ConnectComponent key={Math.random()} name="Max" conID="s1111111" type={2}/>);

    // Test type 3
    component += shallow(<ConnectComponent key={Math.random()} name="Bob" conID="s1111111" type={3}/>);

    // Test type 4
    component += shallow(<ConnectComponent key={Math.random()} name="Soe" conID="s1111111" type={4}/>);
    
    
    expect(component).toMatchSnapshot();
  });