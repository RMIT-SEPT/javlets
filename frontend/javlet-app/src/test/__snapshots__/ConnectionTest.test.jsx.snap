// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectComponent SNAPSHOTt 1`] = `"[object Object][object Object][object Object][object Object][object Object]"`;
